chapter2.adoc

```diff
@@ -134,68 +132,147 @@ The privilege level of the trigger chain is determined by the trigger enabled fo
 
-The dcsr is always accessible in Debug Mode. The access rules for field `prv` and `v` are addressed in subsection <<prvvacc>>. Beside `prv` and `v`, the access condition of remaining fields are listed in the following table.
-When the access conditions are met, they are read/write accessible. When access is disallowed, writes to these fields are ignored, and reads return zero.
+The `sdscratch0` and `sdscratch1` are optional scratch registers are only required if dscratch0 and dscratch1 are implemented.
 
-.Dcsr fields access condition against privilege granted to external debugger
-[options="header"]
-[cols="40%,60%"]
-|============================================
-| Field    |  Access condition
-| debugver |  mdbgen == 1 \|\| sdedbgalw == 1
-| extcause |  mdbgen == 1 \|\| sdedbgalw == 1
-| cetrig   |  mdbgen == 1 
-| ebreakvs |  mdbgen == 1 \|\| sdedbgalw == 1
-| ebreakvu |  mdbgen == 1 \|\| sdedbgalw == 1
-| ebreakm  |  mdbgen == 1
-| ebreaks  |  mdbgen == 1 \|\| sdedbgalw == 1
-| ebreaku  |  mdbgen == 1 \|\| sdedbgalw == 1
-| stepie   |  mdbgen == 1 \|\| sdedbgalw == 1
-| stoptime |  mdbgen == 1
-| mprven   |  mdbgen == 1
-| nmip     |  mdbgen == 1
-|============================================
-
-==== Debug PC (dpc) and Debug Scratch Register (dscratch0 and dscratch1)
+[NOTE]
+In a straightforward implementation, reading or writing any field in `sdcsr` is equivalent to reading or writing the corresponding field in `dcsr`, much like how it works for `sstatus`.
+
+[caption="Register {counter:rimage}: ", reftext="Register {rimage}"]
+[title="Supervisor debug control and status register (sdcsr) for RV32"]
+[id=sdcsr32]
+[wavedrom, ,svg]
+....
+{reg: [
+  {bits:   1, name: 'prv'},
+  {bits:   1, name: '0'},
+  {bits:   1, name: 'step'},
+  {bits:   1, name: '0'},
+  {bits:   1, name: '0'},
+  {bits:   1, name: 'v'},
+  {bits:   3, name: 'cause'},
+  {bits:   1, name: '0'},
+  {bits:   1, name: '0'},
+  {bits:   1, name: 'stepie'},
+  {bits:   1, name: 'ebreaku'},
+  {bits:   1, name: 'ebreaks'},
+  {bits:   1, name: '0'},
+  {bits:   1, name: '0'},
+  {bits:   1, name: 'ebreakvu'},
+  {bits:   1, name: 'ebreakvs'},
+  {bits:   6, name: '0'},
+  {bits:   3, name: 'extcause'},
+  {bits:   1, name: '0'},
+  {bits:   4, name: 'debugver'}
+], config:{lanes: 3, hspace:1024}}
+....
+
+[caption="Register {counter:rimage}: ", reftext="Register {rimage}"]
+[title="Supervisor debug control and status register (sdcsr) for RV64"]
+[id=sdcsr64]
```

Comment:
dcsr is always 32-bit in RISC-V Debug Spec? I do not think we need sdcsr64 in Secure Debug Spec?

Reply:
It doesnt seem compliant to define  a 32-bit CSR when DXLEN=64. The debug spec might just list the lower 32 bits instead of enforcing it to be 32-bit. If a single diagram is clearer, I can merge them into one and denote that it is DXLEN register while only 32 LSBs are used.

Reply:
dcsr is indeed 32 bits wide, regardless of the DXLEN. For example, ireg/iselect CSR are 32-bit regardless of XLEN. 

I think it does not make a difference for implementation. If the target CSR is 32 bits wide (like DCSR or mireg), only the lower 32 bits of the register are affected, and the upper 32 bits of the 64-bit register are ignored during the write operation. The CSR instructions themselves (like CSRW, CSRS, CSRC) are the same, regardless of the size of the CSR. 
For me, it is just an convention. It means the upper 32-bits are not reserved (they will not be used in the future. Anyway, I think it is better to follow the RISC-V Spec like dcsr/ireg/iselect.

Reply:
<img width="571" alt="Screenshot 2024-09-13 at 14 36 57" src="https://github.com/user-attachments/assets/64c46a74-009d-4b0d-8b9a-6f6ac4af3722">
I saw iselect/ireg are also XLEN, probably dcsr is the only exception?

Reply:
No, there are other fixed-width CSRs.  Look at `mvendorid`, for instance.  Using XLEN is most valuable for regs that may hold virtual addresses, or otherwise need to hold more info for RV64 than for RV32.

Reply:
Sorry, I was wrong about iselect/ireg's length. They were not good examples for dcsr fixed-width CSR.